src/analysis_and_optimization/Mir_utils.ml

```diff
@@ -426,6 +426,388 @@ let rec update_expr_ad_levels autodiffable_variables
 and update_idx_ad_levels autodiffable_variables =
   Index.map (update_expr_ad_levels autodiffable_variables)
 
+(**
+ * Modify an index's inner expressions with `op`
+ * @param op a functor returning an expression
+ * @param ind the Index.t to modify
+ *)
+let mod_index op ind =
```

Comment:
This is a `map` function, and Index.t derives `map`, so you should be able to do `map_index`